frontend/src/components/BulkImportDeposits.tsx

```diff
@@ -0,0 +1,203 @@
+        const v = e.target.value;
+        setTextAreaValue(v);
+        if (!v.trim()) {
+            setTextError(null);
+            return;
+        }
+        const parsed = safeParseJson(v);
+        if (!parsed.ok) {
+            setTextError(parsed.error);
+        } else if (!Array.isArray(parsed.value)) {
+            setTextError("JSON is not an array — a single object will be wrapped into an array on import.");
+        } else {
+            setTextError(null);
+        }
+    };
+
+    const onFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+        setFileError(null);
+        setFileContent(null);
+        setFileName(null);
+
+        const files = e.target.files;
+        if (!files || files.length === 0) return;
+
+        const file = files[0];
+        setFileName(file.name);
+
+        const reader = new FileReader();
+        reader.onload = () => {
+            const text = String(reader.result ?? "");
+            const parsed = safeParseJson(text);
+            if (!parsed.ok) {
+                setFileError(parsed.error);
+                return;
+            }
+            setFileContent(Array.isArray(parsed.value) ? parsed.value : [parsed.value]);
+        };
+        reader.onerror = () => setFileError("Failed to read file");
+        reader.readAsText(file);
+    };
+
+    const buildMergedPayload = (): any[] | null => {
+        const arrs: any[] = [];
+        if (fileContent) arrs.push(...fileContent);
+        if (textAreaValue.trim()) {
+            const parsed = safeParseJson(textAreaValue);
+            if (parsed.ok) {
+                arrs.push(...(Array.isArray(parsed.value) ? parsed.value : [parsed.value]));
+            }
+        }
+        return arrs.length > 0 ? arrs : null;
+    };
+
+    const onImportClick = async () => {
+        if (textAreaValue && textError) {
+            toast.current?.show({severity: "error", summary: "Invalid JSON", detail:textError});
+            return;
+        }
+        if (fileError) {
+            toast.current?.show({severity: "error", summary: "Invalid File", detail:textError});
+            return;
+        }
+
+        const payload = buildMergedPayload();
+        if (!payload) {
+            toast.current?.show({ severity: "warn", summary: "Nothing to Import", detail: "Please paste JSON or upload a file." });
+            return;
+        }
+
+        confirmDialog({
+            message: `You are about to import ${payload.length} deposit(s). Proceed?`,
+            header: "Confirm Import",
+            icon: "pi pi-exclamation-triangle",
+            acceptLabel: "Yes",
+            rejectLabel: "No",
+            accept: async () => {
+                setIsSubmitting(true);
+                try {
+                    const resp = await fetch(IMPORT_ENDPOINT, {
+                        method: "POST",
+                        headers: { "Content-Type": "text/plain" },
+                        body: JSON.stringify(payload),
+                    });
```

Comment:
Please add a comment that says to update the request accordingly after #99 is done
P.S. Start the comment with TODO: